web/src/components/tasks/business-formation/business/BusinessStep.tsx

```diff
@@ -1,69 +1,111 @@
+    <hr className="margin-bottom-2 margin-top-0" aria-hidden={true} key={"business-line-3"} />,
+    additionalProvisionsSection(),
+  ];
+
+  const nonprofitOrder: (ReactElement | null)[] = [
+    <MainBusiness key={"main-business"} />,
+    <hr className="margin-bottom-2 margin-top-0" aria-hidden={true} key={"business-line-1"} />,
+    additionalProvisionsSection(),
+    <hr className="margin-bottom-2 margin-top-0" aria-hidden={true} key={"business-line-2"} />,
+    businessPurposeSection(),
+    <hr className="margin-bottom-2 margin-top-0" aria-hidden={true} key={"business-line-3"} />,
+    addressSection(),
+  ];
+
+  const elementsToDisplay = isNonprofit ? nonprofitOrder : defaultOrder;
+
   return (
     <div data-testid="business-step">
-      <MainBusiness />
-      {state.formationFormData.legalType === "limited-partnership" ? (
-        <>
-          <hr className="margin-bottom-2 margin-top-0" aria-hidden={true} />
-          <BusinessFormationTextBox
-            maxChars={400}
-            fieldName={"combinedInvestment"}
-            required={true}
-            title={Config.formation.fields.combinedInvestment.label}
-            contentMd={Config.formation.fields.combinedInvestment.body}
-          />
-          <hr className="margin-bottom-2 margin-top-0" aria-hidden={true} />
-          <BusinessFormationTextBox
-            maxChars={400}
-            fieldName={"withdrawals"}
-            required={true}
-            title={Config.formation.fields.withdrawals.label}
-            contentMd={Config.formation.fields.withdrawals.body}
-          />
-          <hr className="margin-bottom-2 margin-top-0" aria-hidden={true} />
-          <PartnershipRights />
-          <hr className="margin-bottom-2 margin-top-2" aria-hidden={true} />
-          <BusinessFormationTextBox
-            maxChars={400}
-            fieldName={"dissolution"}
-            required={true}
-            title={Config.formation.fields.dissolution.label}
-            contentMd={Config.formation.fields.dissolution.body}
-          />
-        </>
-      ) : (
-        <></>
-      )}
-      <hr className="margin-bottom-2 margin-top-0" aria-hidden={true} />
-      <BusinessFormationTextBox
-        maxChars={300}
-        fieldName={"businessPurpose"}
-        required={false}
-        inputLabel={Config.formation.fields.businessPurpose.label}
-        addButtonText={Config.formation.fields.businessPurpose.addButtonText}
-        title={Config.formation.fields.businessPurpose.label}
-        contentMd={Config.formation.fields.businessPurpose.body}
-      />
-      {state.formationFormData.businessLocationType !== "NJ" &&
-      state.formationFormData.legalType !== "foreign-limited-partnership" ? (
-        <></>
-      ) : (
-        <>
-          <hr className="margin-bottom-2 margin-top-0" aria-hidden={true} />
-          <Provisions />
-        </>
-      )}
+      {elementsToDisplay.map((element, i) => (
+        <Fragment key={i}>{element}</Fragment>
```

Comment:
[pebble] You can do this without the fragment. Supply the key directly to the element.

The current implementation also makes the keys you added in `nonprofitOrder` and `defaultOrder` redundant.

Reply:
Moved HR tags up inside the functions